test/unit/math/fwd/fun/assign_test.cpp

```diff
@@ -73,9 +73,6 @@ TEST(AgradFwdMatrixAssign, eigen_row_vector_fvar_double_shape_mismatch) {
   Matrix<fvar<double>, Dynamic, Dynamic> zzz(3, 1);
   zzz << 1, 2, 3;
   EXPECT_THROW(assign(x, zzz), std::invalid_argument);
-
-  Matrix<fvar<double>, Dynamic, Dynamic> zzzz(1, 3);
-  EXPECT_THROW(assign(x, zzzz), std::invalid_argument);
```

Comment:
This is a tricky one. Size-wise the two objects are the same, but one is declared as a matrix and one a vector. This means that throwing in the math library will be dimension-based, but throwing in the Stan language will be type-dependent (although I'm not sure if ```assign``` is even exposed?). Not sure if that's an important distinction to make somewhere.

Reply:
Indeed. I think there is no reason to have different requirements for `Eigen::Block` and `Eigen::Matrix`. I assume it is better to allow assigning of matrices woth different compile time sizes. Do you think it is prefereble to fail assigments for blocks with different sizes? Or is there a reason for difference in handling of matrices and blocks?